### ISSUE TITLE
handshake-org/hs-tree-data does not exist

### ISSUE THREAD

**Issue by iczero** [handshake-org/hs-airdrop#7]
The handshake-org/hs-tree-data repository does not exist or is private. As a result, hs-airdrop crashes (Invalid checksum) when accessing https://github.com/handshake-org/hs-tree-data/raw/master/faucet.bin.

**Comment by mmcc** [comment]
Running into the same issue. Here's the full error if it helps:

```
$ hs-airdrop [my-address] [my-amount]
Attempting to create proof.
This may take a bit...
Downloading: https://github.com/handshake-org/hs-tree-data/raw/master/faucet.bin...
Error: Invalid checksum: faucet.bin
    at readFile (/Users/[redacted]/.config/yarn/global/node_modules/hs-airdrop/bin/hs-airdrop:86:13)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:182:7)
```

**Comment by karissa** [comment]
Same here

**Comment by karissa** [comment]
Seems like https://github.com/handshake-org/hs-tree-data is a private repo, perhaps?

**Comment by hschoenburg** [comment]
Hi yes it is still private. The repo should be published when the Faucet is complete and can be imported into the airdrop tree. Note "Faucet" is handshake.org and "airdrop" is this on-chain mechanism. The coin distribution is currently transitioning from Faucet (closing down) to Airdrop (coming soon on the next testnet)

**Comment by jamiew** [comment]
Confusing since y'all emailed everyone to try this out. Can we expect another email when it's up and working?

**Comment by hschoenburg** [comment]
Hi @jamiew  yes my apologies for the confusion. The email was to get handshake.org Faucet participants to send in their addresses before the Faucet closes. And then, unnecessarily linked them here.

Given how much confusion the email today caused, I am going to stop using that mailing list. This repo will be published with the launch of the next testnet.

**Comment by actionless** [comment]
> The email was to get handshake.org Faucet participants to send in their addresses before the Faucet closes. 

i was trying to follow the link from the second e-mail but it was leading to this repo when i was clicking to add IRC or e-mail

also on the dashboard it shows 2500 points instead of 4-thousands-something, but i guess it's already too much offtopic for this ticket

**Comment by actionless** [comment]
from the wording of announcement it seemed to me like airdrop is a replacement for faucets for the lately arrived participant, sorry for misunderstanding

**Comment by khoazany** [comment]
> Hi @jamiew yes my apologies for the confusion. The email was to get handshake.org Faucet participants to send in their addresses before the Faucet closes. And then, unnecessarily linked them here.
> 
> Given how much confusion the email today caused, I am going to stop using that mailing list. This repo will be published with the launch of the next testnet.

My address is there on handshake.org but browsing the address hnscan.com results in 0 HNS? Is there something missing?

**Comment by skynode** [comment]
> > Hi @jamiew yes my apologies for the confusion. The email was to get handshake.org Faucet participants to send in their addresses before the Faucet closes. And then, unnecessarily linked them here.
> > Given how much confusion the email today caused, I am going to stop using that mailing list. This repo will be published with the launch of the next testnet.
> 
> My address is there on handshake.org but browsing the same address on hnscan.com results in 0 HNS? Is there something missing?

You're querying the Handshake _testnet_

**Comment by chjj** [comment]
hs-tree-data is now public again for testnet3. It was made private temporarily due to concerns over the privacy aspects of the airdrop (it wasn't usable on testnet2 anyway).

> I was part of the original "faucet". Should I do anything to ensure the transition from "faucet" to "airdrop" goes smooth?

If you were part of the original faucet, your address should be listed in [hs-tree-data/proof.json](https://github.com/handshake-org/hs-tree-data/blob/master/proof.json). Verifying that it's present is a good idea. Let us know if you signed up for the faucet but don't see your address there. Otherwise, feel free to test redeeming it on testnet.

**Comment by actionless** [comment]
> address should be listed in hs-tree-data/proof.json

and what does true/false values mean there?

**Comment by chjj** [comment]
> and what does true/false values mean there?

This will require some explanation...

The merkle tree that `proof.json` proves non-inflation for contains faucet recipients and project sponsors. We wanted a "built-in" fee for the faucet proofs to make it easier for the faucet rewards to go out. This implicit fee is taken out of a user's faucet or sponsor reward. Since the fee is baked into the protocol, the proofs don't need to be signed. This allows someone to publish a faucet proof without needing to retrieve their keys from cold storage.

Project sponsors have `true` set. All it really means on the consensus layer is that they pay a higher implicit fee than the regular faucet. Sponsors pay a 500 coin fee, open source devs pay a 100 coin fee -- those numbers may be subject to change, but the reason they're so high (1/2 and 1/10 of a block reward respectively) is because we can't predict what the fee market will be, but we can safely guess that it won't be that ridiculously high (as an unintentional side effect, early miners will probably get a nice bonus in fees). Sponsors pay more because they have a lot more coins at stake and it only seems logical.

It does seem like a hacky workaround, and there may be alternatives to it. The other option is to just put all of the `proof.json` file into the genesis block, which I think is more annoying in a lot of ways. The third option is to just allow for signed proofs which commit to a fee. This may or may not be bothersome if people have their keys locked up tight. It would also require us to add some HD signing code to this module.

tl;dr: To distinguish project sponsors from faucet recipients.
